backend/src/utils/mod.rs

```
@@ -1,4 +1,175 @@
 //! Collection of general utility functions and common traits.
 //!
 //! This module serves as a repository for small, reusable helper functions
-//! or traits that do not fit into other specific domain modules.
\ No newline at end of file
+//! or traits that do not fit into other specific domain modules.
+
+use serde::{Serialize, Deserialize};
+use bitcoin::secp256k1::PublicKey;
+use std::fmt::{Display, Formatter};
+use crate::errors::LightningError;
+use expanduser::expanduser;
+use lightning::ln::features::NodeFeatures;
+
+/// Represents a node id, either by its public key or alias.
+#[derive(Serialize, Debug, Clone)]
+pub enum NodeId {
+    /// The node's public key.
+    PublicKey(PublicKey),
+    /// The node's alias (human-readable name).
+    Alias(String),
+}
+
+impl NodeId {
+    /// Validates that the provided node id matches the one returned by the backend. If the node id is an alias,
+    /// it will be updated to the one returned by the backend if there is a mismatch.
+    pub fn validate(&self, node_id: &PublicKey, alias: &mut String) -> Result<(), LightningError> {
+        match self {
+            NodeId::PublicKey(pk) => {
+                if pk != node_id {
+                    return Err(LightningError::ValidationError(format!(
+                        "The provided node id does not match the one returned by the backend ({} != {}).",
+                        pk, node_id
+                    )));
+                }
+            },
+            NodeId::Alias(a) => {
+                if alias != a {
+                    log::warn!(
+                        "The provided alias does not match the one returned by the backend ({} != {}).",
+                        a,
+                        alias
+                    )
+                }
+                *alias = a.to_string();
```

Why are we just throwing a warning instead an error here? It doesn't follow the semantics for public key validation above. we shouldn't reassign the Alias if the validation fails. That way you don't need to make `alias` a mutable reference

---

Okay we could throw an error as well, that works too!